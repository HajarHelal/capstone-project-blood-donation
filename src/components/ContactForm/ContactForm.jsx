import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import InputFiled from '../InputField';
import Button from '../Button';

const ContactForm = () => {
  const { t } = useTranslation();
  const data = {
    name: '',
    email: '',
    massege: '',
  };
  const [form, setForm] = useState(data);

  const onChangeHandling = (e) => {
    const { name, value } = e.target;
    setForm({
      ...form,
      [name]: value,
    });
  };

  const onClickHandling = (event) => {
    event.preventDefault();
  };
  return (
    <form className="md:shadow-xl w-10/12 lg:w-8/12 mx-auto  flex flex-col  p-2 sm:p-4 md:p-8 bg-white">
      <div className="flex flex-col md:flex-row mb-8">
        <InputFiled
          placeholder={t('AboutPage.ContectName')}
          onChange={onChangeHandling}
          inputValue={form.name}
          inputName="name"
          Inputtype="text"
        />
        <InputFiled
          placeholder={t('AboutPage.ContectEmail')}
          onChange={onChangeHandling}
          inputValue={form.email}
          inputName="email"
          Inputtype="email"
        />
      </div>
      <textarea
        className=" flex-1 mr-4 p-2 text-base tracking-normal  placeholder-gray-500 border-2 border-gray-300 focus:outline-none "
        rows="6"
        placeholder={t('AboutPage.ContectMessage')}
        value={form.massege}
        name="massege"
        onChange={onChangeHandling}
      />
      <div className="self-center">
        <Button text={t('AboutPage.ContectSend')} onclick={onClickHandling} classFlag={false} />
      </div>
    </form>
  );
};

export default ContactForm;
