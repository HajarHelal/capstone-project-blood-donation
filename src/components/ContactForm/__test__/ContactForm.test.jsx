import React from 'react';
import renderer from 'react-test-renderer';
import ContactForm from '../ContactForm';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer.create(<ContactForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
