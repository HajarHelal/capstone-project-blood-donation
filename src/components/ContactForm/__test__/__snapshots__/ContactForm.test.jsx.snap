// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="md:shadow-xl w-10/12 lg:w-8/12 mx-auto  flex flex-col  p-2 sm:p-4 md:p-8"
>
  <div
    className="flex flex-col md:flex-row mb-8"
  >
    <div
      className="flex-1 flex flex-col  mb-4 "
    >
      <label
        className="flex-1 mb-2 text-sm sm:text-base font-semibold"
        htmlFor="name"
      />
      <input
        className="flex-1  mr-4 p-2 text-sm sm:text-base tracking-normal font-semibold  border-2 border-gray-300 focus:outline-none  bg-white placeholder-gray-800"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="AboutPage.ContectName"
        type="text"
        value=""
      />
    </div>
    <div
      className="flex-1 flex flex-col  mb-4 "
    >
      <label
        className="flex-1 mb-2 text-sm sm:text-base font-semibold"
        htmlFor="email"
      />
      <input
        className="flex-1  mr-4 p-2 text-sm sm:text-base tracking-normal font-semibold  border-2 border-gray-300 focus:outline-none  bg-white placeholder-gray-800"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="AboutPage.ContectEmail"
        type="email"
        value=""
      />
    </div>
  </div>
  <textarea
    className=" flex-1 mr-4 p-2 text-base tracking-normal font-semibold placeholder-gray-800 border-2 border-gray-300 focus:outline-none "
    name="massege"
    onChange={[Function]}
    placeholder="AboutPage.ContectMessage"
    rows="6"
    value=""
  />
  <div
    className="self-center"
  >
    <button
      className="inline-block text-white text-base md:text-lg  text-center md:font-medium bg-primary  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
      onClick={[Function]}
      type="button"
    >
      AboutPage.ContectSend
    </button>
  </div>
</form>
`;
